nautobot/core/api/serializers.py

```diff
@@ -233,6 +271,39 @@ def build_field(self, field_name, info, model_class, nested_depth):
 
         return super().build_field(field_name, info, model_class, nested_depth)
 
+    def _get_natural_key_field_lookups_for_field(self, field_name, natural_key_field_instance):
+        """Extract natural key field lookups for a specific field name.
+
+        Args:
+            field_name (str): The field name to extract lookups for.
+            natural_key_field_instance (dict): The dict containing natural key field values.
+        """
+        data = {}
+        for key, value in natural_key_field_instance.items():
+            if key.startswith(f"{field_name}__"):
+                if isinstance(value, uuid.UUID):
+                    value = str(value)
+                data[key] = value
+        return data
+
+    def to_representation(self, instance):
+        data = super().to_representation(instance)
+        altered_data = {}
+
+        # if `self.natural_keys_values` is not empty means that this is a csv request then add natural_kay key and values
+        if self.natural_keys_values:
+            natural_key_field_instance = [item for item in self.natural_keys_values if item["pk"] == instance.pk][0]
+            for key, value in data.items():
+                if natural_key_field_lookups_for_field := self._get_natural_key_field_lookups_for_field(
+                    key, natural_key_field_instance
+                ):
+                    altered_data |= natural_key_field_lookups_for_field
```

Comment:
Per the Python docs, the `dict |= other_dict` syntax was only added in Python 3.9, so we can't use it yet.